src/main/java/org/aksw/simba/lemming/metrics/metricselection/ComplexityBasedMetricDecider.java

```diff
@@ -0,0 +1,88 @@
+package org.aksw.simba.lemming.metrics.metricselection;
+
+import org.aksw.simba.lemming.ColouredGraph;
+import org.aksw.simba.lemming.metrics.single.SingleValueMetric;
+import org.aksw.simba.lemming.metrics.single.edgetriangles.EdgeNumberOfSimpleTrianglesMetric;
+import org.aksw.simba.lemming.metrics.single.edgetriangles.MultiThreadedNodeNeighborsCommonEdgesMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.EdgeIteratorMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.MultiThreadedNodeNeighborTrianglesMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.NodeIteratorCoreMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.NodeIteratorMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.forward.ForwardMetric;
+
+/**
+ * @author DANISH AHMED on 8/24/2018
+ */
+public class ComplexityBasedMetricDecider {
+    private ColouredGraph graph;
+    private int numEdges;
+    private int numVertices;
+
+    public ComplexityBasedMetricDecider(ColouredGraph graph) {
+        this.graph = graph;
+        this.numEdges = this.graph.getGraph().getNumberOfEdges();
+        this.numVertices = this.graph.getGraph().getNumberOfVertices();
+    }
+
+    public ColouredGraph getGraph() {
+        return graph;
+    }
```

Comment:
Why does the class store a reference to the graph if this is not necessary? It is only using the two numbers which are gathered directly in the constructor.

Reply:
I would even suggest to remove all three private attributes and add the `ColouredGraph` instance as parameter to the two `getMinComlexityForNodeMetric`.